### DISCUSSION TITLE
High CPU Load on PostgeSQL DB  - DT 4.2.0

### DISCUSSION THREAD

**Discussion by ocristian** [DependencyTrack/dependency-track#1027] [Q&A]
Anyone experiencing DT high DB CPU consumption?
We noticed that at least once a week we are having these CPU spikes and we need to restart the DTApiServer….

This is my config:
```
Dependency-Check enabled: No
NPM Audit enabled: Yes
OSS Index enabled: No
Notifications enabled: Slack
VulnDB enabled: No
LDAP enabled: Yes
LDAP synchronization enabled: Yes
LDAP user provisioning enabled: Yes 
Database vendor/version: 
ALPINE_DATABASE_DRIVER=org.postgresql.Driver
ALPINE_DATABASE_DRIVER_PATH=/extlib/postgresql-42.2.18.jar
Java vendor/version:
image: dependencytrack/apiserver:4.2.0
adoptopenjdk/openjdk11:alpine-jre
Host OS vendor/version:
Dockerimage: dependencytrack/apiserver:4.2.0
Distribution: Docker
Proxy server enabled: no
--
Service resources
cpu: 4
memory: 12288
DB resources
vCPU  4
RAM  16 GB
```

Here some metrics from last two weeks:

<img width="874" alt="Screenshot 2021-04-22 at 17 55 25" src="https://user-images.githubusercontent.com/839766/116792793-389ba080-aac3-11eb-93c3-f3d43870eba5.png">

<img width="885" alt="Screenshot 2021-05-01 at 21 11 08" src="https://user-images.githubusercontent.com/839766/116792797-42250880-aac3-11eb-8586-3362f79f883f.png">

Anyone facing the same problem? Any ideas on how to fix it?
Thanks.

**Comment by stevespringett** [comment]
DT mirrors the NVD. The modified feed is only valid for seven days. Therefore DT will pull down all yearly feeds every six days. Expect a spike every six days to account for the hundreds of thousands of records it needs to replicate. Interrupting this process will cause problems. A properly tuned DT and DB server should be able to mirror in under an hour. If it takes longer than that you'll likely need to switch to an instance that is optimized for compute and memory.

**Comment by ocristian** [comment]
Hi, @stevespringett. thanks for the explanation.

I was wondering even after one hour why we don't get back to regular CPU consumption... Looking at the metrics, we can see on 04.19 CPU consumption started to increase and it never returns to normal consumption, only after a DT restart on 04.23.

Here our configuration:
```
DT Service
cpu: 4
memory: 12GB

DT DB
vCPU  4
RAM  16 GB
```

I'll monitor the next few days to see how it behaves.
Let me know if you have any other advice.
Thanks 👍🏻

**Reply by sergioasantiago** [reply]
Hi @stevespringett do you have any idea why the CPU didn't get back to regular consumption, as mentioned by @ocristian ?

**Reply by stevespringett** [reply]
Could be related to #1026

**Reply by sergioasantiago** [reply]
That seems to be exactly the same behaviour that we have. Feel free to close this ticket since it is duplicated with #1026
